gym_xiangqi/xiangqi_game.py

```diff
@@ -126,13 +189,97 @@ def run(self):
 
             self.render()
 
+        self.game_over()
         self.cleanup()
 
     def load_piece_images(self, pieces: list):
+        '''
+        load the image files to the corresponding piece objects
+        '''
         for i in range(1, len(pieces)):
             pieces[i].set_basic_image()
             pieces[i].set_select_image()
 
+    def to_real_coor(self, clicked_coor):
+        '''
+        convert clicked coordinate to real coordinate
+        '''
+        clicked_real_x = (clicked_coor[0] - COOR_OFFSET) // COOR_DELTA
+        clicked_real_y = (clicked_coor[1] - COOR_OFFSET) // COOR_DELTA
+
+        return clicked_real_x, clicked_real_y
+
+    def find_target_piece(self, clicked_coor):
+        '''
+        search for the currently selected piece object
+        if the object is found, return True
+        if not, return False
+        '''
+        # get clicked coordinate
+        clicked_x, clicked_y = clicked_coor
+
+        # find the piece where the clicked coord is within its range
+        for piece in self.agent_piece[1:]:
+
```

Comment:
Let's not have blank lines after `for` and `if`.

Reply:
fixed!